config/application.rb

```diff
@@ -55,7 +55,6 @@ class Application < Rails::Application
 
     config.load_defaults '7.0'
     config.active_record.belongs_to_required_by_default = false
-    config.active_record.legacy_connection_handling = false
```

Comment:
this option was deprecated and is now removed: https://github.com/rails/rails/blob/358d3d3164bef399b19b089f5fed8664a2c3d39e/guides/source/7_1_release_notes.md#removals-5